dns/zone.py

```diff
@@ -1085,3 +1085,45 @@ def from_xfr(xfr, zone_factory=Zone, relativize=True, check_origin=True):
     if check_origin:
         z.check_origin()
     return z
+
+def generate_ptr(zone, relativize=False, zone_factory=Zone):
+    """Generate PTR Zones for a given zone file and returns a dict
+    with origin as key and the zone object as value.
+
+    @param zone: the original zone object
+    @type zone: dns.zone.Zone
+    @param zone_factory: The zone factory to use
+    @type zone_factory: function returning a Zone
+    @raises KeyError: if zone parameter is not a Zone object.
+    @rtype: dict
+    """
+    import dns.reversename
+    import dns.rdtypes.ANY.PTR
+    import copy
+    if not isinstance(zone, Zone):
+            raise KeyError("zone parameter must be a Zone object")
```

Comment:
I used KeyError to be consistent with other functions as _validate_name (line 135, 138) or find_node (line 204) and others.